### PR TITLE
feat: Change name of SimpleCacheClient to CacheClient

### PR DESCRIPTION
**Pull request by cprice404**
As per feedback from dev advocates and users, the name SimpleCacheClient
is a little confusing and we prefer CacheClient.  This commit
implements the name change.

NOTE: we need to retain a deprecated alias using the old name,
so that we don't break backward compatibility.  That will be done
in a separate commit because the diff is too complex to review
when it is done in the same commit.

### CODE REVIEW

File: tests/Integration/Momento.Sdk.Tests/CacheEagerConnectionTest.cs [hunk 1]
```diff
@@ -21,30 +21,30 @@ public class SimpleCacheEagerConnectionTest
...
+    public void CacheClientConstructor_EagerConnection_BadEndpoint()
     {
         var config = Configurations.Laptop.Latest(loggerFactory);
         config = config.WithTransportStrategy(config.TransportStrategy.WithEagerConnectionTimeout(TimeSpan.FromSeconds(2)));
         var authProviderWithBadCacheEndpoint = authProvider.WithCacheEndpoint("cache.cell-external-beta-1.prod.a.momentohq.com:65000");
         // validating that the constructor doesn't fail when the eager connection fails
-        var client = new SimpleCacheClient(config, authProviderWithBadCacheEndpoint, defaultTtl);
+        var client = new CacheClient(config, authProviderWithBadCacheEndpoint, defaultTtl);
     }
 }
```
Comment (malandis) [inline tests/Integration/Momento.Sdk.Tests/CacheEagerConnectionTest.cs:50]: Dropped a newline here

**Review by malandis — COMMENTED** [review]
Looks good; just a small formatting tweak

File: tests/Integration/Momento.Sdk.Tests/CacheEagerConnectionTest.cs (hunk 1 above)
Comment (cprice404) [inline tests/Integration/Momento.Sdk.Tests/CacheEagerConnectionTest.cs:50]:
```suggestion
}

```

Comment (cprice404) [reply tests/Integration/Momento.Sdk.Tests/CacheEagerConnectionTest.cs]: fixed in https://github.com/momentohq/client-sdk-dotnet/pull/406/commits/4857da7e78172a73f148fa8ccadb2872a3df8765

File: src/Momento.Sdk/Momento.Sdk.csproj [hunk 2]
```diff
@@ -57,13 +57,13 @@
...
 	<ProjectExtensions>
-		<MonoDevelop>
-			<Properties>
-				<Policies>
-					<TextStylePolicy TabWidth="4" IndentWidth="4" RemoveTrailingWhitespace="True" NoTabsAfterNonTabs="False" EolMarker="Native" FileWidth="80" TabsToSpaces="True" scope="application/xml" />
-					<XmlFormattingPolicy scope="application/xml" />
-				</Policies>
-			</Properties>
-		</MonoDevelop>
+	  <MonoDevelop>
```
Comment (allenheltondev) [inline src/Momento.Sdk/Momento.Sdk.csproj:60]: These are the kind of updates I like to see!!

**Review by allenheltondev — APPROVED** [review]

**Pull request merged**